File: packages/csharp/src/Anip.Rest/RouteOverride.cs
namespace Anip.Rest;

/// <summary>
/// Allows customizing the path and/or method for a capability route.
/// </summary>
public class RouteOverride
{
    public string? Path { get; }
    public string? Method { get; }

    public RouteOverride(string? path, string? method)
    {
        Path = path;
        Method = method;
    }
}

File: packages/java/anip-rest/src/main/java/dev/anip/rest/RouteOverride.java
package dev.anip.rest;

/**
 * Allows customizing the path and/or method for a capability route.
 */
public class RouteOverride {

    private final String path;
    private final String method;

    public RouteOverride(String path, String method) {
        this.path = path;
        this.method = method;
    }

    public String getPath() {
        return path;
    }

    public String getMethod() {
        return method;
    }
}
